clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -468,17 +469,50 @@ public CreatePartitionsResult createPartitions(Map<String, NewPartitions> newPar
+     * <p>This operation is not transactional so it may succeed for some topics while fail for others.</p>
      *
-     * It may take several seconds after this method returns
+     * <p>It may take several seconds after this method returns
      * success for all the brokers to become aware that the partitions have been created.
      * During this time, {@link AdminClient#describeTopics(Collection)}
-     * may not return information about the new partitions.
-     *
-     * This operation is supported by brokers with version 1.0.0 or higher.
+     * may not return information about the new partitions.</p>
+     *
+     * <p>This operation is supported by brokers with version 1.0.0 or higher.</p>
+     *
+     * <p>The following exceptions can be anticipated when calling {@code get()} on the futures obtained from the
+     * {@link CreatePartitionsResult#values() values()} method of the  returned {@code CreatePartitionsResult}</p>
+     * <ul>
+     *     <li>{@link org.apache.kafka.common.errors.InvalidRequestException}
+     *     if duplicate topics were present in the request</li>
+     *     <li>{@link org.apache.kafka.common.errors.TopicAuthorizationException}
+     *     if the authenticated user is not authorized to alter the topic</li>
+     *     <li>{@link org.apache.kafka.common.errors.InvalidTopicException}
+     *     if the topic is queue for deletion</li>
+     *     <li>{@link org.apache.kafka.common.errors.ReassignmentInProgressException}
+     *     if a partition reassignment is currently in progress</li>
+     *     <li>{@link org.apache.kafka.common.errors.UnknownTopicOrPartitionException}
+     *     if the topic does not exist</li>
+     *     <li>{@link org.apache.kafka.common.errors.InvalidPartitionsException}
+     *     if the requested {@link NewPartitions#totalCount()} is less than the current number of partitions</li>
+     *     <li>{@link org.apache.kafka.common.errors.InvalidReplicaAssignmentException}
+     *     if the requested {@link NewPartitions#assignments()} contain an unknown broker id,
+     *     or contain duplicate broker ids,
+     *     or are incompatible with the increase in the number of partitions or the topic replication factor
+     *     </li>
+     *     <li>{@link org.apache.kafka.common.errors.BrokerNotAvailableException}
+     *     if the requested {@link NewPartitions#assignments()} contain a broker that is currently unavailable
+     *     </li>
+     *     <li>{@link org.apache.kafka.common.errors.InvalidReplicationFactorException}
+     *     if no {@link NewPartitions#assignments()} are given and it is impossible for the broker to assign
+     *     replicas with the topics replication factor
+     *     </li>
+     *     <li>{@link org.apache.kafka.common.errors.TimeoutException}
+     *     if the request was not completed in within the given {@link CreatePartitionsOptions#timeoutMs()}.
+     *     </li>
+     * </ul>
```

Comment:
I agree it's a long list. What exactly would be the criteria for abbreviating? I would suggest anything which depends on cluster state or configuration (rather than various kinds of malformed request, which are more likely to be programming errors). That would result in this list:

 <ul>
     <li>{@link org.apache.kafka.common.errors.AuthorizationException}
     if the authenticated user is not authorized to alter the topic</li>
     <li>{@link org.apache.kafka.common.errors.TimeoutException}
     if the request was not completed in within the given {@link CreatePartitionsOptions#timeoutMs()}.</li>
     <li>{@link org.apache.kafka.common.errors.ReassignmentInProgressException}
     if a partition reassignment is currently in progress</li>
     <li>{@link org.apache.kafka.common.errors.InvalidTopicException}
     if the topic is queue for deletion</li>
     <li>{@link org.apache.kafka.common.errors.BrokerNotAvailableException}
     if the requested {@link NewPartitions#assignments()} contain a broker that is currently unavailable.</li>
     <li>{@link org.apache.kafka.common.errors.InvalidReplicationFactorException}
     if no {@link NewPartitions#assignments()} are given and it is impossible for the broker to assign
     replicas with the topics replication factor.</li>
     <li>Subclasses of {@link org.apache.kafka.common.KafkaException}
     if the request is invalid in some way.</li>
</ul>

which is still long and scary. And in the future we would need to add `PolicyViolationException`.

Reply:
I was thinking that we'd list things that the user may want to handle. Maybe "topic queued for deletion" could perhaps be treated the same as a topic that has been deleted.

I wonder if we could introduce a superclass for the exceptions that depend on cluster state, but that's for the future.

The rest seem fine.